components/fairy-bridge/src/reqwest_backend.rs

```diff
@@ -0,0 +1,119 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::{init_backend, Backend, BackendSettings, FairyBridgeError, Method, Request, Response};
+use std::{sync::Arc, time::Duration};
+
+struct ReqwestBackend {
+    runtime: tokio::runtime::Runtime,
+    client: reqwest::Client,
+}
+
+#[uniffi::export]
+pub fn init_reqwest_backend(settings: BackendSettings) -> Result<(), FairyBridgeError> {
+    // Create a multi-threaded runtime, with 1 worker thread.
+    //
+    // The difference between this and the current thread worker, is that it will manage the worker
```

Comment:
It's a tokio thing.   The runtime always piggybacks on the current thread.  You can spawn tasks on the worker, but nothing happens until you call `block_on`, which then blocks the current thread while the tasks execute.  I thought it would be the right runtime, since we only need one thread, but the execution model is very different.  I'll try to update this comment to make it more clear.